### PR TITLE
Update Kubernetes to v1.24

### PR DESCRIPTION
**Pull request by zaklawrencea**
Kubernetes v1.24 Announcement: https://kubernetes.io/blog/2022/05/03/kubernetes-1-24-release-announcement/
Changelog: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md

## Highlights of Relevant Changes (combined till 1.24.14)
- [ ] `--pod-infra-container-image` flag of kubelet CLI is being deprecated.  [see this](https://github.com/kubernetes/kubernetes/pull/108045) and [Pull request](https://github.bus.zalan.do/teapot/kubernetes-on-ubuntu/pull/299).
- [x] kubectl comes with a lot of auto-completion out of the box now.
- [x] etcd version upgraded to `v3.5.5`. [test clusters](https://github.com/zalando-incubator/kubernetes-on-aws/pull/6276)
- [x] A security vulnerability fixed which allowed a user with ClusterRole of CRD A to be able to read CRD B.
- [x] Improvements to the `csi` plugin which are relevant for us.

There are no immediate action items (1) is set to be removed in `1.26`, so we have some time.

### CODE REVIEW

File: cluster/config-defaults.yaml [hunk 1]
```diff
@@ -593,10 +593,10 @@ tracing_coredns_local_zone_traces_endpoint: ""
 # AMI id given the image name and the Image AWS account owner.
 #
 # [0]: https://github.com/zalando-incubator/cluster-lifecycle-manager/blob/8a9bd1cb2d094038a9e23e646421f8146b48886a/provisioner/template.go#L116
-kuberuntu_image_v1_22_amd64: {{ amiID "zalando-ubuntu-kubernetes-production-v1.22.17-amd64-master-243" "861068367966" }}
-kuberuntu_image_v1_22_arm64: {{ amiID "zalando-ubuntu-kubernetes-production-v1.22.17-arm64-master-243" "861068367966" }}
 kuberuntu_image_v1_23_amd64: {{ amiID "zalando-ubuntu-kubernetes-production-v1.23.17-amd64-master-256" "861068367966" }}
 kuberuntu_image_v1_23_arm64: {{ amiID "zalando-ubuntu-kubernetes-production-v1.23.17-arm64-master-256" "861068367966" }}
+kuberuntu_image_v1_24_amd64: {{ amiID "zalando-ubuntu-focal-20.04-kubernetes-production-v1.24.13-amd64-master-261" "861068367966" }}
+kuberuntu_image_v1_24_arm64: {{ amiID "zalando-ubuntu-focal-20.04-kubernetes-production-v1.24.13-amd64-master-261" "861068367966" }}
```
Comment (mikkeloscar) [inline cluster/config-defaults.yaml]:
```suggestion
kuberuntu_image_v1_24_arm64: {{ amiID "zalando-ubuntu-focal-20.04-kubernetes-production-v1.24.13-arm64-master-261" "861068367966" }}
```

**Comment by demonCoder95** [comment]
Added update to `1.24.14` to align with [the upgrade in the AMI](https://github.bus.zalan.do/teapot/kubernetes-on-ubuntu/pull/282), so the new `1.24.14` based AMI can be tested against `1.24.14` here.

**Comment by demonCoder95** [comment]
Pipeline is stuck in "Create Cluster" as the kubelet on the master node is not running. Need to deprecate and remove the `cni-conf-dir` [flag](https://github.bus.zalan.do/teapot/kubernetes-on-ubuntu/blob/master/runtime/etc/systemd/system/kubelet.service#L21) from the `kubelet` service in the AMI for `1.24`. See the change [here](https://github.com/kubernetes/kubernetes/pull/106907).

Also have to check and remove any other flags that have been deprecated in `1.24` from the kubelet.

**Comment by demonCoder95** [comment]
I think we already have these changes as part of the `containerd` work [here](https://github.com/zalando-incubator/kubernetes-on-aws/pull/6021), once that work is complete, the AMI can be updated to the newer one and these errors should be resolved.

**Comment by demonCoder95** [comment]
Read the CHANGELOG to find relevant changes and updated the PR description with a summary. There are no further action items here except for rolling out the change once `containerd` change is rolled out.

**Comment by demonCoder95** [comment]
e2e passed for the first time! 🎉

**Comment by mikkeloscar** [comment]
:+1:

**Comment by katyanna** [comment]
👍

**Pull request merged**